### ISSUE TITLE
"aria-controls" attribute on buttons

### ISSUE THREAD

**Issue by juliemoynat** [nico3333fr/jquery-accessible-hide-show-aria#22]
Hello,

In order to allow assistive technologies to focus on content, there should be an `aria-controls` attribute on buttons with the "id" of the content to expand as value.

More details: https://www.w3.org/TR/wai-aria-practices/#accordion

**Comment by nico3333fr** [comment]
Hello,

it is already set up, at the beginning of the script:

```
    var attr_control = 'data-controls',
        attr_expanded = 'aria-expanded',
        attr_labelledby = 'data-labelledby',
        attr_hidden = 'data-hidden',
```

However, for "practical reasons" (especially on voice over, to avoid application mode if I remember well - ping @goetsu you will probably be able to explain why), it has been decided not to follow exactly the DP.

;)

**Comment by juliemoynat** [comment]
I understand the reason but another accessibility expert, in an audit, told me to add "aria-controls", so...
I don't like to modify a library but if we have to... :-)

Thank you for your answer.

**Comment by goetsu** [comment]
@juliemoynat you don't have to modify it because it's already something you can choose to add or not in the configuration ;)
@nico3333fr yes you remember well

**Comment by nico3333fr** [comment]
In fact, if I remember well, for Voice over: if it detects certains ARIA attributes, it switches to application mode, and this mode implies more complicated keyboard combinations for the user.

As this component should be simple, this is why we set it up like mentionned before: it works without being more complicated for the users (difference between "accessibilité de terrain" - don't know how to say it in english and "accessibilité des référentiels").

However, your a11y expert and mine's are not wrong, it depends on the environment of the project and the point of view. 

In a near future, I will rewrite this plugin in the same style of coding that the accordion one, allowing to pass a config on the call, and it should fix this issue without updating the code of the plugin.

**Comment by juliemoynat** [comment]
Ok, so I was not so wrong about the fact that we have to modify plugin's code. Adding a variable will not just work, so we have to modify several lines of code (or maybe I'm wrong about the way I should do it...?).

**Comment by nico3333fr** [comment]
At the beginning of the script, you need to replace
```
var attr_control = 'data-controls',
```
by
```
var attr_control = 'aria-controls',
```
that's all ;)

**Comment by juliemoynat** [comment]
Ok!! So cool :-)
Thank you
